main.py

```diff
@@ -0,0 +1,40 @@
+from Modules.Runtime.environment import SAT
+
+if SAT:
+    from Modules.Magnetometer import magnetometer_factory
+    from Modules.Magnetorquer.hBridge import hBridge
+
+from Modules.DataHub.remote import DataHub
+
+import time
+import math
+
+if SAT:
+    sensor = magnetometer_factory.initialize()
+    actuator = hBridge(23, 24, 14, 15)
+
+hub = DataHub("http://rogozin.space/varserver/")
+
+print("Trying to achieve ground station address...")
+req = hub.get(["lab_ip", "lab_port"])
+lab_addr = "http://" + req[0]["content"] + ":" + str(req[1]["content"])
+
+hub = DataHub(lab_addr)
+hub.set_endpoints("/get", "/set")
+
+while True:
+
+    if SAT:
+        axes = sensor.readMagnet()
+        axes = [axes["x"], axes["y"], axes["z"]]
+        magnitude = math.sqrt(sum([x ** 2 for x in axes]))
+
+    else:
+        axes = [0, 0, 0]
+        magnitude = 0
+
+    print("Updating DataHub")
+    hub.set({"sat_mag": [axes, magnitude], "sat_mag_ping": time.time()})
+    print("Updated DataHub")
+
+    time.sleep(0.1)
```

Comment:
The refresh rate of the mangetometer is 8 times a second, if you try to read data from it 10 times a second, things could get nasty os i would suggest changing that to 0.125

Reply:
I actually planned ` hub.set({"sat_mag": [axes, magnitude], "sat_mag_ping": time.time()})` to take additional ~0.1-0.2 sec. Do you really think this is necessary to validate it at the `sleep` level?